functions/go/set-labels/transformer/labels.go

```diff
@@ -0,0 +1,168 @@
+package transformer
+
+import (
+	"fmt"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+// the similar struct esxits in resid.GVK, but there is no function to create an GVK struct without using kyaml
+type FieldSpec struct {
+	Group              string `json:"group,omitempty" yaml:"group,omitempty"`
+	Version            string `json:"version,omitempty" yaml:"version,omitempty"`
+	Kind               string `json:"kind,omitempty" yaml:"kind,omitempty"`
+	Path               string `json:"path,omitempty" yaml:"path,omitempty"` // seperated by /
+	CreateIfNotPresent bool   `json:"create,omitempty" yaml:"create,omitempty"`
+}
+
+func SetLabels(rl *fn.ResourceList) (bool, error) {
+	transformer := LabelTransformer{}
+	if err := transformer.Config(rl.FunctionConfig); err != nil {
+		rl.Results = append(rl.Results, fn.ErrorResult(err))
+		return false, nil
+	}
+	if err := transformer.Transform(rl.Items); err != nil {
+		rl.Results = append(rl.Results, fn.ErrorResult(err))
+		return false, nil
+	}
+	return true, nil
```

Comment:
Suggest also add a result if the set-labels passes. It can be a summary to tell users what happened and how many resources are changed.
FYI, I like your code about displaying the resource level result, but it might be verbose to users. Here's an improvement which allows users to display results by severity level.  Let me know if this makes sense to you   https://github.com/GoogleContainerTools/kpt/issues/2996